src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -199,4 +217,47 @@ private AdvertisementTitleDto buildAdvertisementTitle(Advertisement advertisemen
                 .ownerAvatar(advertisement.getUser().getAvatarImage())
                 .build();
     }
+
+    private AdvertisementDisplayDto buildAdvertisementDisplayDto(Advertisement advertisement) {
+        Location advLocation = advertisement.getLocation();
+        long id = advertisement.getId();
+        Subcategory subcategory = advertisement.getSubcategory();
+        String createdDate = advertisement.getCreated().format(DateTimeFormatter.ofPattern(dateFormat));
+        String age = Optional.ofNullable(advertisement.getAge()).map(AgeRange::getValue).orElse("");
+        return AdvertisementDisplayDto.builder()
+                .advertisementId(id)
+                .images(imageService.getByAdvertisementId(id))
+                .topic(advertisement.getTopic())
+                .description(advertisement.getDescription())
+                .location(convertTo(advLocation, LocationDto.class))
+                .ownerName(getOwnerFullName(advertisement))
+                .ownerAvatar(advertisement.getUser().getAvatarImage())
+                .age(age)
+                .dealType(advertisement.getDealType())
+                .season(advertisement.getSeason())
+                .size(advertisement.getSize())
+                .gender(advertisement.getGender())
+                .phone(getOwnerPhone(advertisement))
+                .readyForOffers(advertisement.isReadyForOffers())
+                .wishesToExchange(advertisement.getWishesToExchange())
+                .subcategory(convertTo(subcategory, SubcategoryDto.class))
+                .category(convertTo(subcategory.getCategory(), CategoryNameDto.class))
+                .createdDate(createdDate)
+                .build();
+    }
+
+    private String getOwnerPhone(Advertisement advertisement) {
+        return advertisement.getUser().getPhones().stream()
+                .findFirst()
+                .map(phone -> String.valueOf(phone.getPhoneNumber()))
```

Comment:
Just pass User here and make it much simpler:
```
var phones = user.getPhones();
if(!phones.isEmpty())
    return String.valueOf(phones.get(0));
return "";
```

Reply:
Changed